packages/flutter/lib/src/widgets/navigator.dart

```diff
@@ -3047,9 +3050,15 @@ class _RouteEntry extends RouteTransitionRecord {
     currentState = _RouteLifecycle.dispose;
   }
 
-  void dispose() {
+  void forcedDispose() {
     assert(currentState.index < _RouteLifecycle.disposed.index);
     currentState = _RouteLifecycle.disposed;
+    route.dispose();
+  }
+
+  void dispose() {
```

Comment:
Maybe leave a comment here or on forcedDispose explaining the difference...

Reply:
e.g. "dispose" waits for all overlayEntries to be unmounted before disposing the route.